### ISSUE TITLE
import gpytorch error

### ISSUE THREAD

**Issue by chalesguo** [cornellius-gp/gpytorch#22]
$ sudo python setup.py install
[sudo] password for ubuntu: 
running install
running bdist_egg
running egg_info
writing dependency_links to gpytorch.egg-info/dependency_links.txt
writing top-level names to gpytorch.egg-info/top_level.txt
writing requirements to gpytorch.egg-info/requires.txt
writing gpytorch.egg-info/PKG-INFO
reading manifest file 'gpytorch.egg-info/SOURCES.txt'
writing manifest file 'gpytorch.egg-info/SOURCES.txt'
installing library code to build/bdist.linux-x86_64/egg
running install_lib
running build_py
copying gpytorch/libfft/__init__.py -> build/lib.linux-x86_64-3.5/gpytorch/libfft
running build_ext
generating cffi module 'build/temp.linux-x86_64-3.5/gpytorch.libfft._libfft.c'
already up-to-date
creating build/bdist.linux-x86_64/egg
creating build/bdist.linux-x86_64/egg/gpytorch
creating build/bdist.linux-x86_64/egg/gpytorch/means
copying build/lib.linux-x86_64-3.5/gpytorch/means/__init__.py -> build/bdist.linux-x86_64/egg/gpytorch/means
copying build/lib.linux-x86_64-3.5/gpytorch/means/mean.py -> build/bdist.linux-x86_64/egg/gpytorch/means
copying build/lib.linux-x86_64-3.5/gpytorch/means/constant_mean.py -> build/bdist.linux-x86_64/egg/gpytorch/means
copying build/lib.linux-x86_64-3.5/gpytorch/gp_model.py -> build/bdist.linux-x86_64/egg/gpytorch
copying build/lib.linux-x86_64-3.5/gpytorch/__init__.py -> build/bdist.linux-x86_64/egg/gpytorch
creating build/bdist.linux-x86_64/egg/gpytorch/random_variables
copying build/lib.linux-x86_64-3.5/gpytorch/random_variables/__init__.py -> build/bdist.linux-x86_64/egg/gpytorch/random_variables
copying build/lib.linux-x86_64-3.5/gpytorch/random_variables/constant_random_variable.py -> build/bdist.linux-x86_64/egg/gpytorch/random_variables
copying build/lib.linux-x86_64-3.5/gpytorch/random_variables/independent_random_variables.py -> build/bdist.linux-x86_64/egg/gpytorch/random_variables
copying build/lib.linux-x86_64-3.5/gpytorch/random_variables/samples_random_variable.py -> build/bdist.linux-x86_64/egg/gpytorch/random_variables
copying build/lib.linux-x86_64-3.5/gpytorch/random_variables/gaussian_random_variable.py -> build/bdist.linux-x86_64/egg/gpytorch/random_variables
copying build/lib.linux-x86_64-3.5/gpytorch/random_variables/batch_random_variables.py -> build/bdist.linux-x86_64/egg/gpytorch/random_variables
copying build/lib.linux-x86_64-3.5/gpytorch/random_variables/random_variable.py -> build/bdist.linux-x86_64/egg/gpytorch/random_variables
copying build/lib.linux-x86_64-3.5/gpytorch/random_variables/categorical_random_variable.py -> build/bdist.linux-x86_64/egg/gpytorch/random_variables
copying build/lib.linux-x86_64-3.5/gpytorch/random_variables/bernoulli_random_variable.py -> build/bdist.linux-x86_64/egg/gpytorch/random_variables
creating build/bdist.linux-x86_64/egg/gpytorch/likelihoods
copying build/lib.linux-x86_64-3.5/gpytorch/likelihoods/__init__.py -> build/bdist.linux-x86_64/egg/gpytorch/likelihoods
copying build/lib.linux-x86_64-3.5/gpytorch/likelihoods/likelihood.py -> build/bdist.linux-x86_64/egg/gpytorch/likelihoods
copying build/lib.linux-x86_64-3.5/gpytorch/likelihoods/gaussian_likelihood.py -> build/bdist.linux-x86_64/egg/gpytorch/likelihoods
copying build/lib.linux-x86_64-3.5/gpytorch/likelihoods/bernoulli_likelihood.py -> build/bdist.linux-x86_64/egg/gpytorch/likelihoods
creating build/bdist.linux-x86_64/egg/gpytorch/lazy
copying build/lib.linux-x86_64-3.5/gpytorch/lazy/__init__.py -> build/bdist.linux-x86_64/egg/gpytorch/lazy
copying build/lib.linux-x86_64-3.5/gpytorch/lazy/kronecker_product_lazy_variable.py -> build/bdist.linux-x86_64/egg/gpytorch/lazy
copying build/lib.linux-x86_64-3.5/gpytorch/lazy/toeplitz_lazy_variable.py -> build/bdist.linux-x86_64/egg/gpytorch/lazy
copying build/lib.linux-x86_64-3.5/gpytorch/lazy/lazy_variable.py -> build/bdist.linux-x86_64/egg/gpytorch/lazy
copying build/lib.linux-x86_64-3.5/gpytorch/module.py -> build/bdist.linux-x86_64/egg/gpytorch
creating build/bdist.linux-x86_64/egg/gpytorch/inference
copying build/lib.linux-x86_64-3.5/gpytorch/inference/__init__.py -> build/bdist.linux-x86_64/egg/gpytorch/inference
creating build/bdist.linux-x86_64/egg/gpytorch/inference/posterior_models
copying build/lib.linux-x86_64-3.5/gpytorch/inference/posterior_models/__init__.py -> build/bdist.linux-x86_64/egg/gpytorch/inference/posterior_models
copying build/lib.linux-x86_64-3.5/gpytorch/inference/posterior_models/gp_posterior.py -> build/bdist.linux-x86_64/egg/gpytorch/inference/posterior_models
copying build/lib.linux-x86_64-3.5/gpytorch/inference/posterior_models/exact_gp_posterior.py -> build/bdist.linux-x86_64/egg/gpytorch/inference/posterior_models
copying build/lib.linux-x86_64-3.5/gpytorch/inference/posterior_models/variational_gp_posterior.py -> build/bdist.linux-x86_64/egg/gpytorch/inference/posterior_models
copying build/lib.linux-x86_64-3.5/gpytorch/inference/inference.py -> build/bdist.linux-x86_64/egg/gpytorch/inference
creating build/bdist.linux-x86_64/egg/gpytorch/functions
copying build/lib.linux-x86_64-3.5/gpytorch/functions/__init__.py -> build/bdist.linux-x86_64/egg/gpytorch/functions
copying build/lib.linux-x86_64-3.5/gpytorch/functions/log_normal_cdf.py -> build/bdist.linux-x86_64/egg/gpytorch/functions
copying build/lib.linux-x86_64-3.5/gpytorch/functions/normal_cdf.py -> build/bdist.linux-x86_64/egg/gpytorch/functions
copying build/lib.linux-x86_64-3.5/gpytorch/functions/dsmm.py -> build/bdist.linux-x86_64/egg/gpytorch/functions
copying build/lib.linux-x86_64-3.5/gpytorch/functions/add_diag.py -> build/bdist.linux-x86_64/egg/gpytorch/functions
creating build/bdist.linux-x86_64/egg/gpytorch/utils
copying build/lib.linux-x86_64-3.5/gpytorch/utils/toeplitz.py -> build/bdist.linux-x86_64/egg/gpytorch/utils
copying build/lib.linux-x86_64-3.5/gpytorch/utils/interpolation.py -> build/bdist.linux-x86_64/egg/gpytorch/utils
copying build/lib.linux-x86_64-3.5/gpytorch/utils/__init__.py -> build/bdist.linux-x86_64/egg/gpytorch/utils
copying build/lib.linux-x86_64-3.5/gpytorch/utils/lincg.py -> build/bdist.linux-x86_64/egg/gpytorch/utils
copying build/lib.linux-x86_64-3.5/gpytorch/utils/fft.py -> build/bdist.linux-x86_64/egg/gpytorch/utils
copying build/lib.linux-x86_64-3.5/gpytorch/utils/lanczos_quadrature.py -> build/bdist.linux-x86_64/egg/gpytorch/utils
copying build/lib.linux-x86_64-3.5/gpytorch/utils/function_factory.py -> build/bdist.linux-x86_64/egg/gpytorch/utils
copying build/lib.linux-x86_64-3.5/gpytorch/utils/kronecker_product.py -> build/bdist.linux-x86_64/egg/gpytorch/utils
copying build/lib.linux-x86_64-3.5/gpytorch/utils/circulant.py -> build/bdist.linux-x86_64/egg/gpytorch/utils
creating build/bdist.linux-x86_64/egg/gpytorch/kernels
copying build/lib.linux-x86_64-3.5/gpytorch/kernels/__init__.py -> build/bdist.linux-x86_64/egg/gpytorch/kernels
copying build/lib.linux-x86_64-3.5/gpytorch/kernels/grid_interpolation_kernel.py -> build/bdist.linux-x86_64/egg/gpytorch/kernels
copying build/lib.linux-x86_64-3.5/gpytorch/kernels/kernel.py -> build/bdist.linux-x86_64/egg/gpytorch/kernels
copying build/lib.linux-x86_64-3.5/gpytorch/kernels/rbf_kernel.py -> build/bdist.linux-x86_64/egg/gpytorch/kernels
copying build/lib.linux-x86_64-3.5/gpytorch/kernels/spectral_mixture_kernel.py -> build/bdist.linux-x86_64/egg/gpytorch/kernels
copying build/lib.linux-x86_64-3.5/gpytorch/kernels/index_kernel.py -> build/bdist.linux-x86_64/egg/gpytorch/kernels
creating build/bdist.linux-x86_64/egg/gpytorch/libfft
copying build/lib.linux-x86_64-3.5/gpytorch/libfft/__init__.py -> build/bdist.linux-x86_64/egg/gpytorch/libfft
copying build/lib.linux-x86_64-3.5/gpytorch/libfft/_libfft.abi3.so -> build/bdist.linux-x86_64/egg/gpytorch/libfft
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/means/__init__.py to __init__.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/means/mean.py to mean.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/means/constant_mean.py to constant_mean.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/gp_model.py to gp_model.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/__init__.py to __init__.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/random_variables/__init__.py to __init__.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/random_variables/constant_random_variable.py to constant_random_variable.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/random_variables/independent_random_variables.py to independent_random_variables.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/random_variables/samples_random_variable.py to samples_random_variable.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/random_variables/gaussian_random_variable.py to gaussian_random_variable.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/random_variables/batch_random_variables.py to batch_random_variables.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/random_variables/random_variable.py to random_variable.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/random_variables/categorical_random_variable.py to categorical_random_variable.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/random_variables/bernoulli_random_variable.py to bernoulli_random_variable.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/likelihoods/__init__.py to __init__.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/likelihoods/likelihood.py to likelihood.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/likelihoods/gaussian_likelihood.py to gaussian_likelihood.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/likelihoods/bernoulli_likelihood.py to bernoulli_likelihood.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/lazy/__init__.py to __init__.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/lazy/kronecker_product_lazy_variable.py to kronecker_product_lazy_variable.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/lazy/toeplitz_lazy_variable.py to toeplitz_lazy_variable.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/lazy/lazy_variable.py to lazy_variable.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/module.py to module.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/inference/__init__.py to __init__.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/inference/posterior_models/__init__.py to __init__.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/inference/posterior_models/gp_posterior.py to gp_posterior.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/inference/posterior_models/exact_gp_posterior.py to exact_gp_posterior.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/inference/posterior_models/variational_gp_posterior.py to variational_gp_posterior.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/inference/inference.py to inference.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/functions/__init__.py to __init__.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/functions/log_normal_cdf.py to log_normal_cdf.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/functions/normal_cdf.py to normal_cdf.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/functions/dsmm.py to dsmm.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/functions/add_diag.py to add_diag.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/utils/toeplitz.py to toeplitz.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/utils/interpolation.py to interpolation.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/utils/__init__.py to __init__.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/utils/lincg.py to lincg.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/utils/fft.py to fft.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/utils/lanczos_quadrature.py to lanczos_quadrature.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/utils/function_factory.py to function_factory.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/utils/kronecker_product.py to kronecker_product.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/utils/circulant.py to circulant.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/kernels/__init__.py to __init__.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/kernels/grid_interpolation_kernel.py to grid_interpolation_kernel.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/kernels/kernel.py to kernel.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/kernels/rbf_kernel.py to rbf_kernel.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/kernels/spectral_mixture_kernel.py to spectral_mixture_kernel.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/kernels/index_kernel.py to index_kernel.cpython-35.pyc
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/libfft/__init__.py to __init__.cpython-35.pyc
creating stub loader for gpytorch/libfft/_libfft.abi3.so
byte-compiling build/bdist.linux-x86_64/egg/gpytorch/libfft/_libfft.py to _libfft.cpython-35.pyc
creating build/bdist.linux-x86_64/egg/EGG-INFO
copying gpytorch.egg-info/PKG-INFO -> build/bdist.linux-x86_64/egg/EGG-INFO
copying gpytorch.egg-info/SOURCES.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying gpytorch.egg-info/dependency_links.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying gpytorch.egg-info/requires.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
copying gpytorch.egg-info/top_level.txt -> build/bdist.linux-x86_64/egg/EGG-INFO
writing build/bdist.linux-x86_64/egg/EGG-INFO/native_libs.txt
zip_safe flag not set; analyzing archive contents...
gpytorch.libfft.__pycache__._libfft.cpython-35: module references __file__
creating 'dist/gpytorch-0.1-py3.5-linux-x86_64.egg' and adding 'build/bdist.linux-x86_64/egg' to it
removing 'build/bdist.linux-x86_64/egg' (and everything under it)
Processing gpytorch-0.1-py3.5-linux-x86_64.egg
removing '/usr/local/lib/python3.5/dist-packages/gpytorch-0.1-py3.5-linux-x86_64.egg' (and everything under it)
creating /usr/local/lib/python3.5/dist-packages/gpytorch-0.1-py3.5-linux-x86_64.egg
Extracting gpytorch-0.1-py3.5-linux-x86_64.egg to /usr/local/lib/python3.5/dist-packages
gpytorch 0.1 is already the active version in easy-install.pth

Installed /usr/local/lib/python3.5/dist-packages/gpytorch-0.1-py3.5-linux-x86_64.egg
Processing dependencies for gpytorch==0.1
Searching for cffi==1.10.0
Best match: cffi 1.10.0
Adding cffi 1.10.0 to easy-install.pth file

Using /usr/local/lib/python3.5/dist-packages
Searching for pycparser==2.18
Best match: pycparser 2.18
Adding pycparser 2.18 to easy-install.pth file

Using /usr/local/lib/python3.5/dist-packages
Finished processing dependencies for gpytorch==0.1
********************************************************************
$ python
Python 3.5.2 (default, Nov 17 2016, 17:05:23) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import gpytorch
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/ubuntu/gpytorch-master/gpytorch/__init__.py", line 3, in <module>
    from .lazy import LazyVariable, ToeplitzLazyVariable
  File "/home/ubuntu/gpytorch-master/gpytorch/lazy/__init__.py", line 2, in <module>
    from .toeplitz_lazy_variable import ToeplitzLazyVariable
  File "/home/ubuntu/gpytorch-master/gpytorch/lazy/toeplitz_lazy_variable.py", line 4, in <module>
    from gpytorch.utils import toeplitz
  File "/home/ubuntu/gpytorch-master/gpytorch/utils/toeplitz.py", line 2, in <module>
    import gpytorch.utils.fft as fft
  File "/home/ubuntu/gpytorch-master/gpytorch/utils/fft.py", line 1, in <module>
    from .. import libfft
  File "/home/ubuntu/gpytorch-master/gpytorch/libfft/__init__.py", line 3, in <module>
    from ._libfft import lib as _lib, ffi as _ffi
ImportError: No module named 'gpytorch.libfft._libfft'

**Comment by gpleiss** [comment]
Looks like you had an old version of the library, which didn't include the libfft stuff.
Try running `python setup.py install --force`.

**Comment by chalesguo** [comment]
Try running python setup.py install --force.  this error still exists.

Remove the directory “gpytorch-0.1-py3.5-linux-x86_64.egg”to reinstall, the problem still exists

**Comment by gpleiss** [comment]
I would upgrade to python 3.6.1

**Comment by lizix7** [comment]
Hi @chalesguo ,
I got the same error with you. 

This is the error message.

> 
ImportError: dlopen(/usr/local/lib/python3.6/site-packages/gpytorch/libfft/_libfft.abi3.so, 2): Symbol not found: _fftwf_destroy_plan
  Referenced from: /usr/local/lib/python3.6/site-packages/gpytorch/libfft/_libfft.abi3.so
  Expected in: flat namespace
 in /usr/local/lib/python3.6/site-packages/gpytorch/libfft/_libfft.abi3.so
>

**Comment by gpleiss** [comment]
Ah - I think I know what's going on.
Make sure you're not running your python script from the `gpytorch` folder.

See: https://github.com/pytorch/pytorch/issues/1633

@lizix7 let me know if this fixes the issue!

**Comment by lizix7** [comment]
Hi @gpleiss ,

Thanks for your response. But I was not running my script from the gpytorch folder.
I used homebrew to install fftw3. I think it was the library linking issue here.

**Comment by gpleiss** [comment]
@chalesguo did you also use homebrew to install fftw3?

**Comment by gpleiss** [comment]
@lizix7 can you copy the output when you run `python setup.py install`? Also, can you include the output when you run `ls gpytorch/libfft`?

**Comment by lizix7** [comment]
Hi @gpleiss 

I used your 
> pip install git+https://github.com/jrg365/gpytorch.git
>

to install gpytorch. So I do not have setup.py there.
This is what I have in the terminal:
`
➜  gpytorch ls libfft
`
`
__init__.py     __pycache__     _libfft.abi3.so
`

Thanks,
Zhixing

**Comment by gpleiss** [comment]
Hmm okay - maybe it's a linking issue then with `fftw`. Can you try unlinking/relinking, or uninstalling and reinstalling?

**Comment by lizix7** [comment]
I have tried both, but neither works.
The location of fftw3 that brew installed is at /usr/local/lib. 
Then I checked the linking of _libfft.abi3.so
`
➜  libfft otool -L _libfft.abi3.so
_libfft.abi3.so:
	/usr/local/opt/fftw/lib/libfftw3.3.dylib (compatibility version 9.0.0, current version 9.7.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.0.0)
`

I will try to create some links if I have time.
Based on this issue #20, @notsimon said it is because fftw3f, but I am not sure with it.

**Comment by jrg365** [comment]
**TL;DR** adding 'fftw3f' to the list of libraries on line 9 of `build.py` might fix the problem, as it did in #20.

I think there is some difference between libfftw and libfftw**f** -- at least, function names appear to be different.

`➜  libfft git:(alpha_release) ldd _libfft.abi3.so`
Reveals:
`        linux-vdso.so.1 =>  (0x00007ffcf65c3000)`
`        libfftw3.so.3 => /usr/lib/x86_64-linux-gnu/libfftw3.so.3 (0x00007f3c77227000)`

And:
`➜  libfft git:(alpha_release) readelf -W --syms /usr/lib/x86_64-linux-gnu/libfftw3.so.3 | grep fftw_destroy_plan`
`   278: 00000000000f6fd0     8 FUNC    GLOBAL DEFAULT   11 dfftw_destroy_plan__`
`   741: 00000000000f5990    50 FUNC    GLOBAL DEFAULT   11 fftw_destroy_plan`
`   824: 00000000000f5e80     8 FUNC    GLOBAL DEFAULT   11 dfftw_destroy_plan_`

However, the same thing on libfftwf as installed by conda's installation of libfftwf is a different story:
`➜  libfft git:(alpha_release) readelf -W --syms /usr/local/anaconda2/lib/libfftw3f.so.3 | grep fftw_destroy_plan`
returns nothing, but (note the extra f!)
`➜  libfft git:(alpha_release) readelf -W --syms /usr/local/anaconda2/lib/libfftw3f.so.3 | grep fftwf_destroy_plan`
`   779: 00000000000f0410    82 FUNC    GLOBAL DEFAULT   11 fftwf_destroy_plan`
`  5069: 00000000000f0410    82 FUNC    GLOBAL DEFAULT   11 fftwf_destroy_plan`

**Comment by jrg365** [comment]
Yeah, if you'd be willing to download the repo and build it yourself with that fix that'd be much appreciated since it'd confirm the fix works! Then we can make a PR that fixes it.

**Comment by lizix7** [comment]
Hi @jrg365 ,

I have downloaded the repo; change `libraries = ['fftw3','fftw3f']` in build.py.
Then I did ` python3 setup.py build` and `python3 setup.py install`.
Next, I ran my regression script. But the problem still exists. 
Then I double checked the `_libfft.abi3.so`.

`➜  libfft otool -L _libfft.abi3.so`
`_libfft.abi3.so:`
`
	/usr/local/opt/fftw/lib/libfftw3.3.dylib (compatibility version 9.0.0, current version 9.7.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.0.0)
`

It seems that the link has not changed. 
Did I miss any steps or execute the wrong command while I was installing gpytorch?
Thanks.

**Comment by jrg365** [comment]
This is so strange. I'm not super familiar with the internals of OS X -- is there an equivalent of `readelf` / `objdump` that would show if `libfftw3.3.dylib` has either of `fftw_destroy_plan` or `fftwf_destroy_plan`? Those are functions that should definitely exist in libfftw.

I'm honestly kind of wondering if something strange is going on with the homebrew install -- on linux, it works both compiled from source (from http://www.fftw.org) and installed from conda. 

I'll look in to this a bit on a mac (maybe tomorrow) and then update.

**Comment by lizix7** [comment]
I did not see fftw_destroy_plan or fftwf_destroy_plan.
This is the first time I use the similar command in MacOs. 
But I have to say this is strange too.

`➜  lib otool -IHGv libfftw3.3.dylib`
```
libfftw3.3.dylib:
Indirect symbols for (__TEXT,__stubs) 19 entries
address            index name
0x000000000012b7ec  2464 ___bzero
0x000000000012b7f2  2465 ___sincos_stret
0x000000000012b7f8  2466 ___stack_chk_fail
0x000000000012b7fe  2470 _abort
0x000000000012b804  2471 _fclose
0x000000000012b80a  2472 _ferror
0x000000000012b810  2473 _fflush
0x000000000012b816  2474 _fopen
0x000000000012b81c  2475 _fprintf
0x000000000012b822  2476 _fread
0x000000000012b828  2477 _free
0x000000000012b82e  2478 _fwrite
0x000000000012b834  2479 _gettimeofday
0x000000000012b83a  2480 _log
0x000000000012b840  2481 _malloc
0x000000000012b846  2482 _memcpy
0x000000000012b84c  2483 _memmove
0x000000000012b852  2484 _qsort
0x000000000012b858  2485 _strcmp
Indirect symbols for (__DATA,__got) 3 entries
address            index name
0x0000000000143000  2467 ___stack_chk_guard
0x0000000000143008  2468 ___stderrp
0x0000000000143010  2469 ___stdoutp
Indirect symbols for (__DATA,__nl_symbol_ptr) 2 entries
address            index name
0x0000000000143018  2486 dyld_stub_binder
0x0000000000143020 ABSOLUTE
Indirect symbols for (__DATA,__la_symbol_ptr) 19 entries
address            index name
0x0000000000143028  2464 ___bzero
0x0000000000143030  2465 ___sincos_stret
0x0000000000143038  2466 ___stack_chk_fail
0x0000000000143040  2470 _abort
0x0000000000143048  2471 _fclose
0x0000000000143050  2472 _ferror
0x0000000000143058  2473 _fflush
0x0000000000143060  2474 _fopen
0x0000000000143068  2475 _fprintf
0x0000000000143070  2476 _fread
0x0000000000143078  2477 _free
0x0000000000143080  2478 _fwrite
0x0000000000143088  2479 _gettimeofday
0x0000000000143090  2480 _log
0x0000000000143098  2481 _malloc
0x00000000001430a0  2482 _memcpy
0x00000000001430a8  2483 _memmove
0x00000000001430b0  2484 _qsort
0x00000000001430b8  2485 _strcmp
Data in code table (7 entries)
offset     length kind
0x00002700     16 JUMP_TABLE32
0x00004b54     40 JUMP_TABLE32
0x00006ce4     24 JUMP_TABLE32
0x000071a0     48 JUMP_TABLE32
0x00044c60     28 JUMP_TABLE32
0x00045028     28 JUMP_TABLE32
0x000c0ba0     24 JUMP_TABLE32
```

**Comment by jrg365** [comment]
Okay, that library file doesn't actually seem to have ANY of the symbols I'd expect in libfftw -- I'll try to get it working with a homebrew installation and see if I can make any progress, but in the meantime I'd suggest either installing from source, using conda as we recommend in the readme, or using fink as recommended on the fftw webpage http://www.fftw.org/install/mac.html

**Comment by lizix7** [comment]
Hi @jrg365 ,

Thanks. 
Homebrew is so much easier and it made me really lazy. :D 
I will try to install it in another way.
Still, have a good night.

**Comment by lizix7** [comment]
@jrg365 Any good news?

**Comment by gpleiss** [comment]
@lizix7 sorry for the slow response. What MacOS version are you using? Also, can you uninstall GPyTorch and reinstall, and paste in the installation output?

**Comment by gpleiss** [comment]
@lizix7 or @chalesguo - if you're using Anaconda, this should be fixed now by #92. We now recommend installing fftw through Anaconda:

```bash
conda install fftw -c conda-forge
```

Give this a shot. If you still have installation issues, feel free to open another github issue.
